README.md

```diff
@@ -1,4 +1,223 @@
+and will update it on each new incoming `CalculatorAction`:
+```kotlin
+val store = coroutineScope.createStore<Int, CalculatorAction>(
+    name = "Calculator",
+    initialState = 0,
+    reducer = { currentState, newAction ->
+        when (newAction) {
+            is CalculatorAction.Add -> currentState + newAction.value
+            is CalculatorAction.Deduct -> currentState - newAction.value
+            is CalculatorAction.Multiply -> currentState * newAction.value
+            is CalculatorAction.Divide -> currentState / newAction.value
+        }
+    }
+)
+```
+
+Where `reducer` is a special function responsible for managing state. A `reducer` specify
+how the state changes in response to actions sent to the store.
+Remember that actions only describe what happened, but don't describe how the application's state changes.
+That is the job of the `reducer`.
+
+Finally you have to subscribe `StateReceiver` to the `store` instance to get the state updates over time:
+```kotlin
+store.subscribe { state -> updateUI(state) }
+```
+Where `StateReceiver` is a function, that will receive each new update of the state inside `store`.
+More then one `StateReceiver` can subscribe to the same `store` instance.
+
+With default parameters (`launchMode = CoroutineStart.LAZY`), on first subscription, `store` will emit `initialState` to it.
+
+On each new UI _intention_, UI implementation just need to send (_dispatch_) it as an action to `store` instance:
+```kotlin
+store.dispatch(CalculatorAction.Add(10))
+// Current state will become 10
+store.dispatch(CalculatorAction.Deduct(1))
+// Current state will become 9
+store.dispatch(CalculatorAction.Add(1))
+// Current state will become 10
+store.dispatch(CalculatorAction.Divide(10))
+// Current state will become 1
+```
+
+All actions will be processed in serialized order
+and on each incoming action `reducer` function will be called to update current state.
+
+### Side effects
+
+Side effect is an interface with exactly one function - it receives incoming
+actions, can get current store state at any time and may emit outgoing actions.
+**So basically it is actions in and actions out.**
+
+**Actions in** to side effects are emitted by store after calling `reducer` function.
+
+**Actions out** are consumed back again by store and trigger calling `reducer` function.
+
+Side effects are used to perform additional `Job` as a reaction on a certain action, for example,
+making an HTTP request, I/O operations, writing to database and so on. Since they run in a `Job` they can run async.
+Each `Job` is scoped to the context of the `store`.
+
+Let's add an side effect that makes a network request each time when action is `CalculatorAction.Add`
+and current state is `9`. If server responds with http code `200` - side effect should emit `CalculatorAction.Deduct(1)`.
+```kotlin
+val sideEffect = object : SideEffect<Int, CalculatorAction> {
+    override val name: String = "network logger"
```

Comment:
> If I understood it corretly SideEffect.name is only used for logging, right?

Correct.

> Maybe the logger can determine the name on demand (i.e. how Jake Wharton's Timber determines the calling function name)

Needs investigation, can be added later as default interface `val name` implementation.

> completely let it to the developer using logger to include this "side effect name" information in the log message.

I would not do that - this will mess with expected log events from the `SideEffect` if developer will use different names.